### PR TITLE
Vendor golint in k/k, and use the vendor version in verify-golint check

### PR DESCRIPTION
**Pull request by krzyzacy**
will eventually move it into the image

/hold
/assign

/sig testing
/priority important-soon
/kind cleanup

### CODE REVIEW

**Review by cblecker — CHANGES_REQUESTED** [review]
I'm not sure I like this for multiple reasons:
- Forces an update to golint in gopath, strictly because golint isn't in PATH.
- Doesn't address issues like `${GOPATH}/bin` isn't in PATH
- Doesn't pin the package, so linting could be inconsistent

We should vendor golint and use that. We've done similar things with gazelle, kazel, cfssl, etc

**Comment by krzyzacy** [comment]
@cblecker fork in third_party and build from src if not found?

**Comment by krzyzacy** [comment]
I thought k/k deps were generated from all packages? How would I manually add it?

**Comment by cblecker** [comment]
Godep docs: https://github.com/kubernetes/community/blob/master/contributors/devel/godep.md

Basic steps:
1. Restore godeps: `hack/run-in-gopath.sh hack/godep-restore.sh`
1. go get the new dep: `hack/run-in-gopath.sh go get -du golang.org/x/lint/golint`
1. Add to the [additional required deps](https://github.com/kubernetes/kubernetes/blob/e1552d8ef9043604aae6b08b2742c9d4f3f281bc/hack/godep-save.sh#L55-L67) section of hack/godep-save.sh
1. Save godeps: `hack/run-in-gopath.sh hack/godep-save.sh`
1. Modify hack/verify-golint.sh to install and use golint from vendor. Use [something similar](https://github.com/kubernetes/kubernetes/blob/e1552d8ef9043604aae6b08b2742c9d4f3f281bc/hack/update-bazel.sh#L27-L33) to what we have in hack/update-bazel.sh

**Comment by krzyzacy** [comment]
my bash is not the best 🤷‍♂️

**Comment by krzyzacy** [comment]
https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/72255/pull-kubernetes-verify-podutil/15 - golint passed there, will fix remote branch name in a separate PR.

/unassign
/assign @cblecker @BenTheElder @ixdy @dims

File: hack/verify-golint.sh [hunk 1]
```diff
@@ -24,9 +24,13 @@ source "${KUBE_ROOT}/hack/lib/init.sh"
 kube::golang::verify_go_version
 
 if ! which golint > /dev/null; then
```
Comment (BenTheElder) [inline hack/verify-golint.sh]: seems like we should always use the vendored one to match ci? 🤔 

(I forget if k/k uses this pattern elsewhere, but always using the copy from vendor has worked well for other projects at least...)

**Comment by BenTheElder** [comment]
Thanks Sen! one comment

File: hack/verify-golint.sh (hunk 1 above)
Comment (krzyzacy) [reply hack/verify-golint.sh]: make sense

**Comment by dims** [comment]
/lgtm

Thanks @krzyzacy

File: hack/verify-golint.sh (hunk 1 above)
Comment (BenTheElder) [reply hack/verify-golint.sh]: update the PR title? :-)

**Review by BenTheElder — APPROVED** [review]
/lgtm
Thank you!

(You should update the PR title to reflect that we always use the vendored copy now)

I believe @cblecker's points have been addressed.

**Review by ixdy — APPROVED** [review]

**Comment by krzyzacy** [comment]
updated PR title, will make cherrypicks (I think this golint version supports all active k8s branches, we'll see)

/hold cancel

**Review by cblecker — COMMENTED** [review]
/lgtm
/approve

**Review by cblecker — APPROVED** [review]

**Pull request merged**